master/usage/ipv6.md

```diff
@@ -1,33 +1,147 @@
+  should be configured to assign only IPv6 addresses.
+- The steps and setup here have not been tested against an existing IPv4
+  cluster and are intended only for new clusters.
+
+### Prerequisites
+
+#### Host prerequisites
+
+- Each Kubernetes host must have an IPv6 address that is reachable from
+  the other hosts as well as an IPv4 address.
+- Each host must have the sysctl setting `net.ipv6.conf.all.forwarding`
+  setting it to `1`.  This ensures both Kubernetes service traffic
+  and Calico traffic is forwarded appropriately.
+- Each host must have a default IPv6 route.
+
+#### Kubernetes components prerequisites
+
+Kubernetes components must be configured to operate with IPv6.
+To enable IPv6, set the following flags.
+- For kube-apiserver:
+  - The `--bind-address` or `--insecure-bind-address` flags should be set
+    to the appropriate IPv6 address or `::` for all IPv6 addresses on
+	the host.
+  - The `--advertise-address` flag should be set to the IPv6 address that
+    nodes should use to access the kube-apiserver.
+  - The `--service-cluster-ip-range` flag should be set to an IPv6 CIDR
+    that will be used for the Service IPs, the DNS service address must
+	be in this range.
+- For kube-controller-manager:
+  - The `--master` flag should be set with the IPv6 address where the
+    kube-apiserver can be accessed.
+  - If the `--allocate-node-cidrs` flag is set then it is necessary to set
+    `--node-cidr-mask-size`. These are necessary when using host-local IPAM,
+    but if using calico-ipam is is easier to remove both flags as they are
+    not needed.
+  - The `--cluster-cidr` flag must be set to match the {{site.prodname}}
+    IPv6 IPPool.
+- For kube-scheduler:
+  - The `--master` flag should be set with the IPv6 address where the
+    kube-apiserver can be accessed.
+- For kubelet:
+  - The `--address` flag should be set to the appropriate IPv6 address
+    or `::` for all IPv6 addresses.
+  - The `--cluster-dns` flag should be set to the IPv6 address that will
+    be used for the service DNS, this must be in the range used for
+	`--service-cluster-ip-range`.
+  - The `--node-ip` flag should be set to the IPv6 address of the node.
+- For kube-proxy:
+  - The `--bind-address` flag should be set to the appropriate IPv6
+    address or `::` for all IPv6 addresses on the host.
+  - The `--master` flag should be set with the IPv6 address where the
+    kube-apiserver can be accessed.
+  - The `--cluster-cidr` flag must be set to match the {{site.prodname}}
+    IPv6 IPPool.
+
+### Enabling IPv6 support in Calico
+
+To enable IPv6 support when installing {{site.prodname}} follow the
+steps below.
+
+1. Download the {{site.prodname}} manifest you wish to update for IPv6
+   deployment and save it as `calico.yaml`.
+1. If the ipam section in the `cni_network_config` in the `calico.yaml` file
+   has `"type": "calico-ipam"` then it should be modified to have
+   [IPv4 assigning disabled and IPv6 assigning
```

Comment:
nit: probably better to say "should be modified to disable IPv4 assignments and enable IPv6 assignments."